site/content/how-to/monitoring/troubleshooting.md

```diff
@@ -65,54 +79,312 @@ LAST SEEN   TYPE      REASON              OBJECT
+        proxy_set_header Upgrade "$http_upgrade";
+        proxy_set_header Connection "$connection_upgrade";
+        proxy_http_version 1.1;
+        proxy_pass http://default_coffee_80$request_uri;
+    }
+
+    location = /coffee {
+        proxy_set_header Host "$gw_api_compliant_host";
+        proxy_set_header X-Forwarded-For "$proxy_add_x_forwarded_for";
+        proxy_set_header Upgrade "$http_upgrade";
+        proxy_set_header Connection "$connection_upgrade";
+        proxy_http_version 1.1;
+        proxy_pass http://default_coffee_80$request_uri;
+    }
+
+    location / {
+        return 404 "";
+    }
+
+}
+upstream default_coffee_80 {
+    random two least_conn;
+    zone default_coffee_80 512k;
+
+    server 10.244.0.13:8080;
+}
+```
+
+Some key things to note here:
+
+1. A new `server` block is created with the hostname of the HTTPRoute. When a request is sent to this hostname, it will be handled by this `server` block.
+2. Within the `server` block, three new `location` blocks are added for *coffee*, each with distinct prefix and exact paths. Requests directed to the *coffee* application with a path prefix `/coffee/hello` will be managed by the first location block, while those with an exact path `/coffee` will be handled by the second location block. Any other requests not recognized by the server block for this hostname will default to the third location block, returning a 404 Not Found status.
+3. Each `location` block has headers and directives that configure the NGINX proxy to forward requests to the `/coffee` path correctly, preserving important client information and ensuring compatibility with the upstream server.
+4. The `upstream` block in the given NGINX configuration defines a group of backend servers and configures how NGINX should load balance requests among them.
+
+Now let's check the behaviour when curl request is sent to the `coffee` application:
+
+Matches location /coffee/ block
+
+```shell
+curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/coffee/hello
+Handling connection for 8080
+Server address: 10.244.0.13:8080
+Server name: coffee-56b44d4c55-hwpkp
+Date: 13/Jun/2024:22:51:52 +0000
+URI: /coffee/hello
+Request ID: 21fc2baad77337065e7cf2cd57e04383
+```
+
+Matches location = /coffee block
+
+```shell
+curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/coffee
+Handling connection for 8080
+Server address: 10.244.0.13:8080
+Server name: coffee-56b44d4c55-hwpkp
+Date: 13/Jun/2024:22:51:40 +0000
+URI: /coffee
+Request ID: 4d8d719e95063303e290ad74ecd7339f
+```
+
+Matches location / block
+
+```shell
+curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/
+Handling connection for 8080
+<html>
+<head><title>404 Not Found</title></head>
+<body>
+<center><h1>404 Not Found</h1></center>
+<hr><center>nginx/1.25.4</center>
+</body>
+```
+
+**Warning** The configuration may change in future releases.
```

Comment:
I think we want to use a hugo callout: https://github.com/nginxinc/nginx-gateway-fabric/tree/main/site#using-hugo-shortcodes